### PR TITLE
Add metrics for multi region

### PR DESCRIPTION
**Pull request by johnnyaug**
Following #2327, adding a metric to the BI counting the AWS clients created for each region.

~In order to access the stats collector from the relevant place, I put it on the go-context passed to the block adapter (don't like it - open to suggestions).~

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -356,11 +357,7 @@ const floatSumTolerance = 1e-6
 
 // GetCommittedTierFSParams returns parameters for building a tierFS.  Caller must separately
 // build and populate Adapter.
-func (c *Config) GetCommittedTierFSParams(ctx context.Context) (*pyramidparams.ExtParams, error) {
-	adapter, err := factory.BuildBlockAdapter(ctx, c)
-	if err != nil {
-		return nil, fmt.Errorf("build block adapter: %w", err)
-	}
+func (c *Config) GetCommittedTierFSParams(ctx context.Context, adapter block.Adapter) (*pyramidparams.ExtParams, error) {
```
Comment (itaiad200) [inline pkg/config/config.go:360]: Is the context still needed?

File: pkg/config/config.go [hunk 2]
```diff
@@ -10,14 +10,15 @@ import (
 	"strings"
 	"time"
 
+	"github.com/treeverse/lakefs/pkg/block"
```
Comment (itaiad200) [inline pkg/config/config.go:13]: arrange deps

**Review by itaiad200 — APPROVED** [review]

**Comment by arielshaqed** [comment]
This will probably conflict with #2377, make sure to rebase to `master` HEAD before merging.

File: cmd/lakefs/cmd/run.go [hunk 3]
```diff
@@ -113,8 +107,13 @@ var runCmd = &cobra.Command{
 				Error("Block adapter NOT SUPPORTED for production use")
 		}
 		metadata := stats.NewMetadata(ctx, logger, blockstoreType, authMetadataManager, cloudMetadataProvider)
-		bufferedCollector := stats.NewBufferedCollector(metadata.InstallationID, blockStore.RuntimeStats, cfg)
-
+		bufferedCollector := stats.NewBufferedCollector(metadata.InstallationID, cfg)
+		// init block store
+		blockStore, err := factory.BuildBlockAdapter(context.WithValue(ctx, block.ContextKeyStatsCollector, bufferedCollector), cfg)
```
Comment (nopcoder) [inline cmd/lakefs/cmd/run.go:112]: Talked f2f over passing the stats collector as a parameter to the factory.
The factory will pass the collector using the specific adapter options.
The adapter can define the interface of how to send stats and not directly use the stats package.

File: pkg/block/s3/adapter.go [hunk 4]
```diff
@@ -93,7 +94,13 @@ func WithTranslator(t block.UploadIDTranslator) func(a *Adapter) {
...
 }
 
-func NewAdapter(awsSession *session.Session, opts ...func(a *Adapter)) *Adapter {
+func WithStatsCollector(s stats.Collector) func(a *Adapter) {
+	return func(a *Adapter) {
+		a.clients = a.clients.WithStatsCollector(s)
+	}
+}
+
+func NewAdapter(_ context.Context, awsSession *session.Session, opts ...func(a *Adapter)) *Adapter {
```
Comment (nopcoder) [inline pkg/block/s3/adapter.go:103]: now we can remove the ctx we added in this pr

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/block/s3/adapter.go (hunk 4 above)
Comment (johnnyaug) [reply pkg/block/s3/adapter.go:103]: oops

**Review by nopcoder — APPROVED** [review]

**Pull request merged**